File: AiRPGEnv/gui.cpp
#include "Gui.h"

using namespace std;
using namespace cv;

const int hero_radius = 1;

class Circle 
{
	public:

		int x;
		int y;
		int radius;

		Circle(int &x, int &y, int &radius) : x(x), y(y), radius(radius) {};
};

void GUI::ClearWindow()
{
	Mat game_map = imread(img4map);
	current_game = game_map.clone();
}

void GUI::ShowWindow()
{
	imshow(window_name, current_game);
	waitKey(1);
}

GUI::GUI(const char* &img4map, const char* &window_name) :img4map(img4map), window_name(window_name)
{
	ClearWindow();

	namedWindow(window_name, WINDOW_NORMAL);
	
	ShowWindow();
}

void GUI::DrawHero(int x, int y)
{
	circle(
		current_game,
		Point(x, y),
		hero_radius,
		Scalar(0, 0, 255),
		FILLED
	);
}

void GUI::DrawMonster(int x, int y)
{
	circle(
		current_game,
		Point(x, y),
		hero_radius,
		Scalar(255, 0,0 ),
		FILLED
	);
}

int GUI::GetKeyPressed()
{
	return waitKeyEx(50);
}

File: Server_py/gui.py
import copy
import cv2

import Server_py.global_vars as gv


class GUI:
    def __init__(self, img4map, window_name):
        self.current_game = None
        self.orig = cv2.imread(img4map)
        self.window_name = window_name

        self.clear_window()
        gv.map_size = (self.current_game.shape[1], self.current_game.shape[0])

    def clear_window(self):
        self.current_game = copy.deepcopy(self.orig)

    def show_window(self):
        cv2.namedWindow(self.window_name, cv2.WINDOW_NORMAL)
        cv2.imshow(self.window_name, self.current_game)
        cv2.waitKey(1)

    def is_blue(self, x, y):
        blue = False
        self.clear_window()
        hsv_im = cv2.cvtColor(self.current_game, cv2.COLOR_BGR2HSV)

        color = hsv_im[y, x, 0]
        if 138 > color > 78:
            blue = True
        return blue

    def draw_hero(self, x, y):
        cv2.circle(
            self.current_game,
            (int(x), int(y)),
            1,
            (0, 0, 255)
        )

    def draw_monster(self, x, y):
        cv2.circle(
            self.current_game,
            (int(x), int(y)),
            1,
            (255, 0, 0)
        )
